### PR TITLE
CFD fund transfer improvements

### PR DESCRIPTION
**Pull request by ravali-deriv**
CFD fund transfer improvements
card : https://app.clickup.com/t/20696747/CFDS-3380 

In this card I have worked on separating action and validation part of code & created different functions inorder to avoid redundant lines of code
I have added code to use automation accounts for CFD fundtransfer for different FIAT currencies

Screenshot 2024-07-08 at 4 57 26 PM
Screenshot 2024-07-08 at 8 26 54 PM

### CODE REVIEW

File: cypress/e2e/wip/CFD/testcases/MT5/fundTransferUSD_USD.cy.js [hunk 1]
```diff
@@ -1,21 +1,51 @@
...
 
   it('MT5 account deposit, USD to Derived SVG', () => {
     mt5_tradershub.selectRealAccount()
-    fundTransferPageObjects.mt5Deposit('US Dollar', 'Derived SVG')
+    fundTransferPageObjects.mt5Deposit(fiatAccount, mt5Account)
+    fundTransferPageObjects.validateBalance(
+      mt5Account,
+      fiatAccount.name,
+      'deposit'
+    )
```
Comment (rajendra-deriv) [inline cypress/e2e/wip/CFD/testcases/MT5/fundTransferUSD_USD.cy.js:39]: You can create a common method. 
'`MT5 account deposit, USD to Derived SVG`' and '`MT5 account withdrawal, Derived SVG to USD`' are doing the same operation in QATEST-37180 and QATEST-37216.

File: cypress/e2e/wip/CFD/testcases/MT5/fundTransferGBP_USD.cy.js [hunk 2]
```diff
@@ -1,21 +1,51 @@
...
   it('MT5 account deposit, GBP to Derived SVG', () => {
     mt5_tradershub.selectRealAccount()
-    fundTransferPageObjects.mt5Deposit('Pound Sterling', 'Derived SVG')
+    fundTransferPageObjects.mt5Deposit(fiatAccount, mt5Account)
+    fundTransferPageObjects.validateBalance(
+      mt5Account,
+      fiatAccount.name,
+      'deposit'
+    )
   })
```
Comment (rajendra-deriv) [inline cypress/e2e/wip/CFD/testcases/MT5/fundTransferGBP_USD.cy.js:40]: You can create a common method.
'MT5 account deposit, USD to Derived SVG' and 'MT5 account withdrawal, Derived SVG to USD' are doing the same operation in QATEST-37180 and QATEST-37216.

File: cypress/e2e/wip/CFD/testcases/MT5/fundTransferEUR_USD.cy.js [hunk 3]
```diff
@@ -1,21 +1,51 @@
...
 
   it('MT5 account deposit, EUR to Derived SVG', () => {
     mt5_tradershub.selectRealAccount()
-    fundTransferPageObjects.mt5Deposit('Euro', 'Derived SVG')
+    fundTransferPageObjects.mt5Deposit(fiatAccount, mt5Account)
+    fundTransferPageObjects.validateBalance(
+      mt5Account,
+      fiatAccount.name,
+      'deposit'
+    )
```
Comment (rajendra-deriv) [inline cypress/e2e/wip/CFD/testcases/MT5/fundTransferEUR_USD.cy.js:39]: You can create a common method.
'MT5 account deposit, USD to Derived SVG' and 'MT5 account withdrawal, Derived SVG to USD' are doing the same operation in QATEST-37180, QATEST-37204 and QATEST-37216.

File: cypress/e2e/wip/CFD/testcases/MT5/fundTransferEUR_USD.cy.js [hunk 4]
```diff
@@ -1,21 +1,51 @@
...
+
 describe('QATEST-37204 - MT5 Deposit & withdrawal for different fiat currencies (EUR-USD)', () => {
   beforeEach(() => {
     cy.clearCookies()
     cy.clearLocalStorage()
-    cy.c_login()
     cy.c_visitResponsive('/appstore/traders-hub', 'large')
+    //cy.c_login()
+    cy.c_login({ user: 'cfdfundTransferEUR' })
+    cy.wait(2000) // need to wait until mt5 is loaded on tradershub
```
Comment (rajendra-deriv) [inline cypress/e2e/wip/CFD/testcases/MT5/fundTransferEUR_USD.cy.js:28]: Avoid using hard coded waits as the page might load or might not load and if page doesn't load in 2secs the next step might fail. Instead you can use
`cy.findByTestId('dt_balance_text_container', { timeout: 40000 })`
or 
```
cyLocator: () =>
        cy.findByText('Your identity verification failed because:'),
      timeout: 3000,
      maxRetries: 5,
    })
```
This will dynamically wait.

File: cypress/e2e/wip/CFD/testcases/MT5/fundTransferAUD_USD.cy.js [hunk 5]
```diff
@@ -1,21 +1,51 @@
...
+
+describe('QATEST-37216 - MT5 Deposit & withdrawal for different fiat currencies (AUD-USD)', () => {
   beforeEach(() => {
     cy.clearCookies()
     cy.clearLocalStorage()
-    cy.c_login()
     cy.c_visitResponsive('/appstore/traders-hub', 'large')
+    //cy.c_login()
+    cy.c_login({ user: 'cfdfundTransferAUD' })
+    cy.wait(2000)
```
Comment (rajendra-deriv) [inline cypress/e2e/wip/CFD/testcases/MT5/fundTransferAUD_USD.cy.js:28]: Avoid using hard coded waits as the page might load or might not load and if page doesn't load in 2secs the next step might fail. Instead you can use
`cy.findByTestId('dt_balance_text_container', { timeout: 40000 })`
or

```
cyLocator: () =>
        cy.findByText('Your identity verification failed because:'),
      timeout: 3000,
      maxRetries: 5,
    })
```
This will dynamically wait.

File: cypress/e2e/wip/CFD/testcases/MT5/fundTransferAUD_USD.cy.js [hunk 6]
```diff
@@ -1,21 +1,51 @@
...
 
   it('MT5 account deposit, AUD to Derived SVG', () => {
     mt5_tradershub.selectRealAccount()
-    fundTransferPageObjects.mt5Deposit('Australian Dollar', 'Derived SVG')
+    fundTransferPageObjects.mt5Deposit(fiatAccount, mt5Account)
+    fundTransferPageObjects.validateBalance(
+      mt5Account,
+      fiatAccount.name,
+      'deposit'
+    )
```
Comment (rajendra-deriv) [inline cypress/e2e/wip/CFD/testcases/MT5/fundTransferAUD_USD.cy.js:39]: You can create a common method.
'MT5 account deposit, USD to Derived SVG' and 'MT5 account withdrawal, Derived SVG to USD' are doing the same operation in QATEST-37180, QATEST-37204 and QATEST-37216.

File: cypress/e2e/wip/CFD/Pages/mt5PageObjects.js [hunk 7]
```diff
@@ -11,6 +11,41 @@ class mt5_tradershub {
     this.tradershubDropDown.click()
     this.selectReal.click()
   }
+  checkMT5AccountisPresent = () => {
+    cy.findByTestId('dt_traders_hub')
+      .findByText('Deriv MT5')
+      .should('be.visible')
+    cy.wait(2000)
```
Comment (rajendra-deriv) [inline cypress/e2e/wip/CFD/Pages/mt5PageObjects.js:18]: Avoid using hard coded waits as the page might load or might not load and if page doesn't load in 2secs the next step might fail. Instead you can use
cy.findByTestId('dt_balance_text_container', { timeout: 40000 })
or

cyLocator: () =>
        cy.findByText('Your identity verification failed because:'),
      timeout: 3000,
      maxRetries: 5,
    })
This will dynamically wait.

File: cypress/e2e/wip/CFD/Pages/fundTransferPageObjects.js [hunk 8]
```diff
@@ -1,345 +1,225 @@
...
+      .should('be.visible')
+      .click()
+    cy.findByText('Your funds have been transferred', {
+      exact: true,
+    }).should('be.visible')
+    cy.get(locators.fundTransferLocators.closeButton)
+      .should('be.visible')
+      .contains('Close')
+      .click()
+    cy.wait(2000) // to load mt5 balance on tradershub after transfer
```
Comment (rajendra-deriv) [inline cypress/e2e/wip/CFD/Pages/fundTransferPageObjects.js:92]: Avoid using hard coded waits as the page might load or might not load and if page doesn't load in 2secs the next step might fail. Instead you can use
cy.findByTestId('dt_balance_text_container', { timeout: 40000 })
or

cyLocator: () =>
        cy.findByText('Your identity verification failed because:'),
      timeout: 3000,
      maxRetries: 5,
    })
This will dynamically wait.

File: cypress/e2e/wip/CFD/Pages/fundTransferPageObjects.js
Comment (bilalmajid-deriv) [inline cypress/e2e/wip/CFD/Pages/fundTransferPageObjects.js:1]: Please help to move this and any similar files to cypress/support/pageobjects/Mt5 the e2e folder should not have these it should only have the tests.

Can do it in this PR or a separate PR as well. Ultimately we will be updating this to not use PageObjects.

Comment (ravali-deriv) [reply cypress/e2e/wip/CFD/Pages/fundTransferPageObjects.js:1]: we will take this up in the next PR @bilalmajid-deriv, we will export all the files at once

**Review by rajendra-deriv — APPROVED** [review]

**Comment by rajendra-deriv** [comment]
LGTM

**Review by meeyun-deriv — APPROVED** [review]
LGTM
